[CI] CrossClusterSearchUnavailableClusterIT testSkipUnavailableDependsOnSeeds failing

**Build Scans:**
- [elasticsearch-periodic #6738 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/7poengriinvzq)
- [elasticsearch-periodic #6703 / openjdk17_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/mkowwfnt2e4ge)

**Reproduction Line:**
```
./gradlew ":qa:ccs-unavailable-clusters:javaRestTest" --tests "org.elasticsearch.search.CrossClusterSearchUnavailableClusterIT.testSkipUnavailableDependsOnSeeds" -Dtests.seed=8621B408D6E00BB2 -Dtests.locale=is-Latn-IS -Dtests.timezone=America/Montserrat -Druntime.java=17 -Dtests.fips.enabled=true
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.CrossClusterSearchUnavailableClusterIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testSkipUnavailableDependsOnSeeds),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
#1:
	in [elasticsearch[node0][transport_worker][T#12]][testSearchSkipUnavailable]
	org.elasticsearch.transport.OutboundHandler.lambda$sendResponse$1(OutboundHandler.java:158)
	org.elasticsearch.core.IOUtils.close(IOUtils.java:71)
	org.elasticsearch.core.IOUtils.close(IOUtils.java:87)
	org.elasticsearch.core.IOUtils.close(IOUtils.java:63)
	org.elasticsearch.core.Releasables.close(Releasables.java:73)
	org.elasticsearch.core.Releasables.close(Releasables.java:41)
	org.elasticsearch.core.Releasables$3.close(Releasables.java:129)
	org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:217)
	org.elasticsearch.transport.OutboundHandler$1.onResponse(OutboundHandler.java:257)
	org.elasticsearch.transport.OutboundHandler$1.onResponse(OutboundHandler.java:253)
	org.elasticsearch.transport.netty4.Netty4Utils.lambda$addLis
[truncated]
```

**Issue Reasons:**
- [8.x] 2 failures in test testSkipUnavailableDependsOnSeeds (1.3% fail rate in 159 executions)
- [8.x] 2 failures in step openjdk17_checkpart1_java-fips-matrix (33.3% fail rate in 6 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic (33.3% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/qa/ccs-unavailable-clusters/src/javaRestTest/java/org/elasticsearch/search/CrossClusterSearchUnavailableClusterIT.java
+++ b/qa/ccs-unavailable-clusters/src/javaRestTest/java/org/elasticsearch/search/CrossClusterSearchUnavailableClusterIT.java
@@ -91,7 +91,12 @@ public class CrossClusterSearchUnavailableClusterIT extends ESRestTestCase {
                 EsExecutors.DIRECT_EXECUTOR_SERVICE,
                 SearchShardsRequest::new,
                 (request, channel, task) -> {
-                    channel.sendResponse(new SearchShardsResponse(List.of(), List.of(), Collections.emptyMap()));
+                    var searchShardsResponse = new SearchShardsResponse(List.of(), List.of(), Collections.emptyMap());
+                    try {
+                        channel.sendResponse(searchShardsResponse);
+                    } finally {
+                        searchShardsResponse.decRef();
+                    }
                 }
             );
             newService.registerRequestHandler(
@@ -119,7 +124,12 @@ public class CrossClusterSearchUnavailableClusterIT extends ESRestTestCase {
                         builder.add(node);
                     }
                     ClusterState build = ClusterState.builder(clusterName).nodes(builder.build()).build();
-                    channel.sendResponse(new ClusterStateResponse(clusterName, build, false));
+                    var clusterStateResponse = new ClusterStateResponse(clusterName, build, false);
+                    try {
+                        channel.sendResponse(clusterStateResponse);
+                    } finally {
+                        clusterStateResponse.decRef();
+                    }
                 }
             );
             newService.start();

